classes/core/SettingsBuilder.php

```diff
@@ -0,0 +1,341 @@
+<?php
+
+/**
+ * @file classes/core/SettingsBuilder.php
+ *
+ * Copyright (c) 2014-2024 Simon Fraser University
+ * Copyright (c) 2000-2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class SettingsBuilder
+ *
+ * @brief The class that extends Eloquent's builder to support settings tables for Models
+ */
+
+namespace PKP\core;
+
+use Closure;
+use Illuminate\Contracts\Database\Query\ConditionExpression;
+use Illuminate\Contracts\Database\Query\Expression;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Query\Builder as QueryBuilder;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Str;
+use stdClass;
+
+class SettingsBuilder extends Builder
+{
+    /**
+     * Get the hydrated models without eager loading.
+     *
+     * @param  array|string  $columns
+     *
+     * @return \Illuminate\Database\Eloquent\Model[]|static[]
+     */
+    public function getModels($columns = ['*'])
+    {
+        $rows = $this->getModelWithSettings($columns);
+        $returner = $this->model->hydrate(
+            $rows->all()
+        )->all();
+
+        return $returner;
+    }
+
+    /**
+     * Update records in the database, including settings
+     *
+     * @return int
+     */
+    public function update(array $values)
+    {
+        // Separate Model's primary values from settings
+        [$settingValues, $primaryValues] = collect($values)->partition(
+            fn (mixed $value, string $key) => in_array(Str::camel($key), $this->model->getSettings())
+        );
+
+        // Don't update settings if they aren't set
+        if ($settingValues->isEmpty()) {
+            return parent::update($primaryValues->toArray());
+        }
+
+        $newQuery = clone $this->query;
+
+        if ($primaryValues->isNotEmpty()) {
+            $count = parent::update($primaryValues->toArray());
+        }
+
+        // TODO Eloquent transforms attributes to snake case, find and override instead of transforming here
+        $settingValues = $settingValues->mapWithKeys(
+            fn (mixed $value, string $key) => [Str::camel($key) => $value]
+        );
+
+        $us = $this->model->getSettingsTable();
+        $primaryKey = $this->model->getKeyName();
+
+
+        $sql = $this->buildUpdateSql($settingValues, $us, $newQuery);
+
+        // Build a query for update
+        $settingCount = DB::table($us)->whereIn($us . '.' . $primaryKey, $newQuery->select($primaryKey))
+            ->update([$us . '.setting_value' => DB::raw($sql)]);
+
+        return $count ? $count + $settingCount : $settingCount; // TODO Return the count of updated setting rows?
```

Comment:
I don't know what is better to return here since we updating to tables at once?

Reply:
@Vitaliy-1  Instead of returning a single summed count, could we return an associative array with separate counts for the primary table and the settings table? 
we can also use database transaction ('DB::transaction') to ensure that both updates either succeed or fail together to maintain data integrity.

Reply:
Opened an issue: https://github.com/pkp/pkp-lib/issues/10487